src/mbgl/util/mapbox.cpp

```diff
@@ -50,6 +50,11 @@ std::string normalizeStyleURL(const std::string& url, const std::string& accessT
     }
 
     std::vector<std::string> pathname = getMapboxURLPathname(url);
+
+    if (pathname.size() < 3) {
+        return url;
```

Comment:
If a mapbox: URL has an unexpected format, wouldn’t this be a good opportunity to provide an assertion message to that effect? Otherwise you’ll wind up with a 404 but no other details. It’ll still be a crash but a helpful one.